cocotb/share/lib/vpi/VpiImpl.cpp

```diff
@@ -127,6 +127,7 @@ gpi_objtype_t to_gpi_objtype(int32_t vpitype)
         case vpiIntegerNet:
             return GPI_INTEGER;
 
+        case vpiConstant:
```

Comment:
We should probably figure out which type of constant it is and return that type here. In any case, the handle will be a `ConstantObject`, but calling `get_type` on the handle will be wrong.

Reply:
`vpiParameter` also supports `vpiConstType`, so this would be a good opportunity to deprecate `GPI_PARAMETER` and use the appropriate types for both cases.

Reply:
Yeah, GPI_PARAMETER is not a great solution and I'd like to remove it, but I was considering this PR may come in with 1.4 since it's a fairly innocuous change. That could be another PR if you are up to it.

Reply:
This would probably be a part of a bigger refactor to handle where we deprecate `ConstantObject` and add a constant check to the instances.